src/types/ErrorRate.ts

```diff
@@ -0,0 +1,171 @@
+import { NA, RequestHealth } from './Health';
+import { RateHealthConfig, ToleranceConfig } from './ServerConfig';
+import { serverConfig } from '../config/ServerConfig';
+import { ThresholdStatus, getRequestErrorsStatus, RATIO_NA } from './Health';
+
+export interface ErrorRatio {
+  errorRatio: { status: ThresholdStatus; protocol?: string };
+  inboundErrorRatio: { status: ThresholdStatus; protocol?: string };
+  outboundErrorRatio: { status: ThresholdStatus; protocol?: string };
+}
+
+export interface Rate {
+  requestRate: number;
+  errorRate: number;
+  errorRatio: number;
+}
+
+export const regexProtocol = {
+  grpc: new RegExp(/^[1-9]$|^1[0-6]$/),
+  /* eslint-disable-next-line */
+  http: new RegExp(/^[4-5]\d\d$/)
+};
+
+export const checkExpr = (expr: string, value: string): boolean => {
+  if (expr !== '' && value !== '') {
+    const reg = new RegExp(expr);
+    return reg.test(value);
+  }
+  return false;
+};
+
+const getConfig = (ns: string, name: string, kind: string): RateHealthConfig | undefined => {
+  for (let rate of serverConfig.healthConfig.rate) {
+    if (checkExpr(rate.namespace, ns) && checkExpr(rate.name, name) && checkExpr(rate.kind, kind)) {
```

Comment:
As I commented in the other PR, whenever possible I think it's better to use explicit string matching than regex matching especially when there's only a few allowed values.

That said, regex could also be optimized a little bit. serverConfig is assumed not to change, so all regex could be instanciated once from config (`new RegExp()`), and only testing would be performed here.

Reply:
Done